python/ray/import_thread.py

```diff
@@ -157,13 +157,13 @@ def f():
 
     def fetch_and_execute_function_to_run(self, key):
         """Run on arbitrary function on the worker."""
-        driver_id, serialized_function = self.redis_client.hmget(
-            key, ["driver_id", "function"])
+        (driver_id, serialized_function,
+         run_on_other_drivers) = self.redis_client.hmget(
+             key, ["driver_id", "function", "run_on_other_drivers"])
 
-        if (self.worker.mode in [ray.SCRIPT_MODE, ray.SILENT_MODE]
+        if (run_on_other_drivers == "False"
+                and self.worker.mode in [ray.SCRIPT_MODE, ray.SILENT_MODE]
```

Comment:
There are some issues with this if statement. In particular, `run_on_other_drivers` can be `b"False"`. I think there are other issues as well. See https://github.com/ray-project/ray/pull/2769